datastore/importers/nmdb-import-systeminfo/Parser.test.cpp

```diff
@@ -0,0 +1,276 @@
+      DHCP Enabled:    Yes
+      [01]: 192.168.0.2
+      [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+      )STR", //Missing ipaddresses label
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name: Ethernet
+      DHCP Enabled:    Yes
+      IP address(es)
+      )STR" //Missing Ipaddress data
+    };
+    for (const auto& test : testsOk) {
+      BOOST_TEST(nmdp::test(test.c_str(), parserRule, blank),
+                "Parse rule 'networkCard': " << test);
+    }
+    for (const auto& test: testsFail) {
+      BOOST_TEST(!nmdp::test(test.c_str(), parserRule, blank),
+                "Parse rule 'networkCard fails': " << test);
+    }
+    tp.data.network_cards.clear();
+  }
+  { // Test Data
+    const auto& parserRule {tp.networkCard};
+    std::string out;
+    std::string test {
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name: Ethernet
+      DHCP Enabled:    Yes
+      IP address(es)
+      [01]: 192.168.0.2
+      [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+      )STR",
+    };
+    BOOST_TEST(nmdp::testAttr(test.c_str(), parserRule, out, blank),
+              "Parse rule 'networkCard': " << test);
+    const auto debugString { tp.data.network_cards["realtek pcie gbe family controller"].toDebugString()};
+    BOOST_TEST(tp.data.network_cards["realtek pcie gbe family controller"].isValid());
+    for (const auto& ipAddr : tp.data.network_cards["realtek pcie gbe family controller"].getIpAddresses()) {
+      BOOST_TEST(ipAddr.isValid());
+    }
+    nmdp::testInString(debugString, "name: realtek pcie gbe family controller");
+    nmdp::testInString(debugString, "mediaType: ethernet");
+    nmdp::testInString(debugString, "ipAddress: 192.168.0.2/32");
+    nmdp::testInString(debugString, "ipAddress: 2001:db8:85a3::8a2e:370:7334/128");
+  }
+}
+
+BOOST_AUTO_TEST_CASE(testhotfixes)
+{
+  TestParser tp;
+  {
+    const auto& parserRule {tp.hotfixes};
+    std::string test {
+      R"STR(Hotfix(s):           10 Hotfix(s) Installed.
+      [01]: KB000001
+      [02]: KB000002
+      [03]: KB000003
+      [04]: KB000003
+      [05]: KB000004
+      [06]: KB000005
+      [07]: KB000006
+      [08]: KB000007
+      [09]: KB000008
+      [10]: KB000009
+      )STR"
+
+    };
+
+    BOOST_TEST(nmdp::test(test.c_str(), parserRule, blank, false),
+                "Parse rule 'hotfixes': " << test);
+    //Leaving this as a quantity check due to hotfix being the actual 
+    //data being parsed
+    BOOST_TEST(10 == tp.data.hotfixes.size()); 
+
+    tp.data.hotfixes.clear();
+  }
+  {
+    //Test Buffer Exhaustion Parsing
+    const auto& parserRule {tp.hotfixes};
+    std::string test {
+      // Simulate buffer exhaustion
+      R"STR(Hotfix(s):           3 Hotfix(s) Installed.
+      [01]: KB000001
+      [02]: KB000002
+      [03]: KB000003
+      [04]: K
+      )STR"
+    };
+    std::vector<std::string> out;
+    BOOST_TEST(nmdp::testAttr(test.c_str(), parserRule, out, blank, false),
```

Comment:
Now that I've looked at the whole logic, this test should fully consume the input because you actually want to get rid of the partial line.  Otherwise the rest of the rules won't match correctly (e.g., `networkCards`) when they should because there is still bad data in the buffer.
```suggestion
    BOOST_TEST(nmdp::testAttr(test.c_str(), parserRule, out, blank),
```

Reply:
The only problem is I get an error because it couldn't parse the text properly? The false is what fixed the error because it allowed the parser to fail the parse on that line which is expected?

Reply:
We talked about this out of band, just putting summary notes for reference.

The failure is expected in the `hotfix` rule, but the `hotfixes` rule needs to completely consume the section labeled "Hotfix(s)".  So, your rules are wrong.

That is, the `systemInfo` rule contains `... > -hotfixes > -networkCards > qi::omit[...`, so if the read buffer still has the next data as `[04]: K`, then `networkCards` won't match and ultimately it will drop all data after the `[04]: K` line since the `qi::omit[]` rule is designed to consume till the end of the input.

Reply:
Okay addressed this and the tester is working properly. Removed ignoredLine.